### PR TITLE
jwk misc updates

### PR DESCRIPTION
**Pull request by zjma**
### Description

- Fix `jwks::initialize` visibility (so governance can call).
- Run smoke tests with full nodes to ensure we are not breaking them.
- Do not start `JWKManager` if self is not in the validator set.
- When certifying a JWK update, explicitly reject signatures from non-validator-set-members.
- new `jwks.move` API `remove_issuer_from_observed_jwks()` that might be useful in disabling an issuer.
- UT coverage of `ObservedJWKs` operations.
- Fix a flaky UT that I got the luck to run into.
- dead code clean up.
- Logs.
- RPC DDOS prevention.

### CODE REVIEW

File: crates/aptos-jwk-consensus/src/epoch_manager.rs [hunk 1]
```diff
@@ -153,11 +153,17 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
...
+            .address_to_validator_index()
+            .get(&self.my_addr)
+            .copied();
+
         info!("[JWK] start_new_epoch: new_epoch={}", epoch_state.epoch);
 
         let features = payload.get::<Features>().unwrap_or_default();
 
-        if features.is_enabled(FeatureFlag::JWK_CONSENSUS) {
+        if let (true, Some(_my_index)) = (features.is_enabled(FeatureFlag::JWK_CONSENSUS), my_index) {
```
Comment (danielxiangzl) [inline crates/aptos-jwk-consensus/src/epoch_manager.rs:166]: `if features.is_enabled(FeatureFlag::JWK_CONSENSUS) && my_index.is_some()`?

File: aptos-move/framework/aptos-framework/sources/jwks.move [hunk 2]
```diff
@@ -187,6 +187,10 @@ module aptos_framework::jwks {
     /// Remove an OIDC provider from the `SupportedOIDCProviders` resource.
     /// Can only be called in a governance proposal.
     /// Returns the old config URL of the provider, if any, as an `Option`.
+    ///
+    /// NOTE: this only stops validators from watching the provider and generate updates to `ObservedJWKs`.
+    /// It does NOT touch `ObservedJWKs` or `Patches`.
```
Comment (danielxiangzl) [inline aptos-move/framework/aptos-framework/sources/jwks.move:192]: what is patch?

File: crates/aptos-jwk-consensus/src/observation_aggregation/mod.rs [hunk 3]
```diff
@@ -57,6 +57,7 @@ impl BroadcastStatus<JWKConsensusMsg> for Arc<ObservationAggregationState> {
             author == sender,
             "adding peer observation failed with mismatched author",
         );
+        ensure!(self.epoch_state.verifier.get_voting_power(&author).is_some(), "adding peer observation failed with illegal signer");
```
Comment (zekun000) [inline crates/aptos-jwk-consensus/src/observation_aggregation/mod.rs:60]: I "think" this is redundant, the signature verification returns some similar errors

**Review by zekun000 — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 4]
```diff
@@ -1003,7 +1003,7 @@ module aptos_framework::account {
         // Maul the signature and make sure the call would fail
         let invalid_signature = ed25519::signature_to_bytes(&sig);
         let first_sig_byte = vector::borrow_mut(&mut invalid_signature, 0);
-        *first_sig_byte = *first_sig_byte + 1;
+        *first_sig_byte = *first_sig_byte ^ 1;
```
Comment (alinush) [inline aptos-move/framework/aptos-framework/sources/account.move:1006]: Is ^ XOR? What was wrong with +?

**Review by alinush — APPROVED** [review]
Please clarify the ^ change though.

File: aptos-move/framework/aptos-framework/sources/account.move (hunk 4 above)
Comment (zjma) [reply aptos-move/framework/aptos-framework/sources/account.move:1006]: yes it's XOR. byte + 1 may overflow. but XOR won't.

**Pull request merged**